tsl/test/sql/data_node.sql

```diff
@@ -777,3 +779,161 @@ DROP DATABASE :DN_DBNAME_3;
 DROP DATABASE :DN_DBNAME_4;
 DROP DATABASE :DN_DBNAME_5;
 DROP DATABASE :DN_DBNAME_6;
+
+
+-----------------------------------------------
+-- Test alter_data_node()
+-----------------------------------------------
+SELECT node_name, database, node_created, database_created, extension_created FROM add_data_node('data_node_1', host => 'localhost', database => :'DN_DBNAME_1');
+SELECT node_name, database, node_created, database_created, extension_created FROM add_data_node('data_node_2', host => 'localhost', database => :'DN_DBNAME_2');
+SELECT node_name, database, node_created, database_created, extension_created FROM add_data_node('data_node_3', host => 'localhost', database => :'DN_DBNAME_3');
+
+GRANT USAGE ON FOREIGN SERVER data_node_1, data_node_2, data_node_3 TO :ROLE_1;
+SET ROLE :ROLE_1;
+
+CREATE TABLE hyper1 (time timestamptz, location int, temp float);
+CREATE TABLE hyper2 (LIKE hyper1);
+CREATE TABLE hyper3 (LIKE hyper1);
+CREATE TABLE hyper_1dim (LIKE hyper1);
+
+SELECT create_distributed_hypertable('hyper1', 'time', 'location', replication_factor=>1);
+SELECT create_distributed_hypertable('hyper2', 'time', 'location', replication_factor=>2);
+SELECT create_distributed_hypertable('hyper3', 'time', 'location', replication_factor=>3);
+SELECT create_distributed_hypertable('hyper_1dim', 'time', chunk_time_interval=>interval '2 days', replication_factor=>3);
+
+SELECT setseed(1);
+INSERT INTO hyper1
+SELECT t, (abs(timestamp_hash(t::timestamp)) % 3) + 1, random() * 30
+FROM generate_series('2022-01-01 00:00:00'::timestamptz, '2022-01-05 00:00:00', '1 h') t;
+
+INSERT INTO hyper2 SELECT * FROM hyper1;
+INSERT INTO hyper3 SELECT * FROM hyper1;
+INSERT INTO hyper_1dim SELECT * FROM hyper1;
+
+-- create view to see the data nodes and default data node of all
+-- chunks
+CREATE VIEW chunk_query_data_node AS
+SELECT ch.hypertable_name, format('%I.%I', ch.chunk_schema, ch.chunk_name)::regclass AS chunk, ch.data_nodes, fs.srvname default_data_node
+	   FROM timescaledb_information.chunks ch
+	   INNER JOIN pg_foreign_table ft ON (format('%I.%I', ch.chunk_schema, ch.chunk_name)::regclass = ft.ftrelid)
+	   INNER JOIN pg_foreign_server fs ON (ft.ftserver = fs.oid)
+	   ORDER BY 1, 2;
+
+SELECT * FROM chunk_query_data_node;
+
+-- test alter_data_node permissions
+\set ON_ERROR_STOP 0
+-- must be owner to alter a data node
+SELECT * FROM alter_data_node('data_node_1', available=>false);
+SELECT * FROM alter_data_node('data_node_1', port=>8989);
+\set ON_ERROR_STOP 1
+
+-- query some data from all hypertables to show its working before
+-- simulating the node being down
+SELECT time, location FROM hyper1 ORDER BY time LIMIT 1;
+SELECT time, location FROM hyper2 ORDER BY time LIMIT 1;
+SELECT time, location FROM hyper3 ORDER BY time LIMIT 1;
+SELECT time, location FROM hyper_1dim ORDER BY time LIMIT 1;
+
+-- simulate a node being down by renaming the database for
```

Comment:
Maybe I am missing it, but do we have a test where try to mark all the nodes as unavailable?

Reply:
I can add one, but what is it that you'd like to see from that test?

Reply:
Nothing concrete, seems like an edge case